modules/outbound-command-event-handler/test/integration/application/process_sdk_outbound_bulk_quotes_request.test.ts

```diff
@@ -0,0 +1,327 @@
+    expect(partyInfoRequestedDomainEvents.length).toEqual(bulkRequest.individualTransfers.length);
+
+    // Get the randomly generated transferIds for the callback
+    const randomGeneratedTransferIds = await bulkTransactionEntityRepo.getAllIndividualTransferIds(bulkTransactionId);
+
+    // Simulate the domain handler sending the command handler PProcessPartyInfoCallback messages
+    // for each individual transfer
+    const processPartyInfoCallbackMessageData1: IProcessPartyInfoCallbackCmdEvtData = {
+      bulkId: bulkTransactionId,
+      content: {
+        transferId: randomGeneratedTransferIds[0],
+        partyResult: {
+          party: {
+              partyIdInfo: {
+                  partyIdType: 'MSISDN',
+                  partyIdentifier: '123456',
+                  fspId: 'receiverfsp'
+              }
+          },
+          currentState: 'COMPLETED'
+        },
+      },
+      timestamp: Date.now(),
+      headers: []
+    }
+    const processPartyInfoCallbackMessageData2: IProcessPartyInfoCallbackCmdEvtData = {
+      bulkId: bulkTransactionId,
+      content: {
+        transferId: randomGeneratedTransferIds[1],
+        partyResult: {
+          party: {
+              partyIdInfo: {
+                  partyIdType: 'MSISDN',
+                  partyIdentifier: '123456',
+                  fspId: 'receiverfsp'
+              }
+          },
+          currentState: 'COMPLETED'
+        },
+      },
+      timestamp: Date.now(),
+      headers: []
+    }
+
+    const processPartyInfoCallbackMessageObjOne = new ProcessPartyInfoCallbackCmdEvt(processPartyInfoCallbackMessageData1);
+    await producer.sendCommandEvent(processPartyInfoCallbackMessageObjOne);
+    const processPartyInfoCallbackMessageObjTwo = new ProcessPartyInfoCallbackCmdEvt(processPartyInfoCallbackMessageData2);
+    await producer.sendCommandEvent(processPartyInfoCallbackMessageObjTwo);
+    await new Promise(resolve => setTimeout(resolve, messageTimeout));
+
+    // Simulate the domain handler sending the command handler ProcessSDKOutboundBulkPartyInfoRequestComplete message
+    const processSDKOutboundBulkPartyInfoRequestCompleteCommandEventData : IProcessSDKOutboundBulkPartyInfoRequestCompleteCmdEvtData = {
+      bulkId: bulkTransactionId,
+      timestamp: Date.now(),
+      headers: []
+    }
+    const processSDKOutboundBulkPartyInfoRequestCompleteCommandEventObj = new ProcessSDKOutboundBulkPartyInfoRequestCompleteCmdEvt(
+      processSDKOutboundBulkPartyInfoRequestCompleteCommandEventData
+    );
+    await producer.sendCommandEvent(processSDKOutboundBulkPartyInfoRequestCompleteCommandEventObj);
+    await new Promise(resolve => setTimeout(resolve, messageTimeout));
+
+    // Check that the global state to be DISCOVERY_COMPLETED
+    const bulkState = await bulkTransactionEntityRepo.load(bulkTransactionId);
+    expect(bulkState.state).toBe('DISCOVERY_COMPLETED');
```

Comment:
No we don't technically need the assertions. Since the flow of most of the messages require previous ones to be run it made sense to just copy/paste the the code from the handlers that are run previously in the test as well.

Let me know if it adds to much bloat. I can remove them.

Reply:
I think it will be good to separate the logic to build the state and the logic for assertions. If it is not too much of work, my preference will be to remove the assertions where not needed.

Reply:
done.